content/docs/setup/kubernetes/multicluster-install/gateways/index.md

```diff
@@ -0,0 +1,242 @@
+    app: bar
+{{< /text >}}
+
+The namespaces and services need to be replicated in each cluster in order
+to enable DNS resolution and connectivity. So, apply the configuration in `cluster1-svc.yaml` to
+`cluster2` and the configuration in `cluster2-svc.yaml` to `cluster1`.
+
+## Route traffic for remote services to remote cluster gateway
+
+Creating a service object for a remote service enables DNS resolution for
+remote services inside the application code. The network traffic from the
+application will be destined for the cluster VIP associated with the
+Kubernetes service. In order for Envoy to route traffic, it needs to know
+the set of remote endpoints for the service to which traffic must be routed
+to.  Using the example depicted in the diagram, we created a service
+declaration for `bar.ns2` in `cluster1`. Now, add an endpoint to this
+service in `cluster1` using the Istio service entry as follows:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: remote-endpoint-for-bar-in-cluster1
+spec:
+  hosts:
+  - bar.ns2.svc.cluster.local
+  ports:
+  # Add all ports from bar service
+  - number: 80
+    name: http
+    protocol: HTTP
+  - number: 9090
+    name: tcp
+    protocol: TCP
+  resolution: STATIC
+  location: MESH_INTERNAL
+  endpoints:
+  - address: <IPofCluster2IngressGateway>
+    ports:
+      http: 15443 #Map everything to 15443 of ingress gateway
+      tcp: 15443
+{{< /text >}}
+
+The configuration above will result in all traffic in `cluster1` for
+`bar.ns2.svc.cluster.local` for any port to be routed to the endpoint
+`<IPofCluster2IngressGateway>:15443` over a mTLS connection.
+
+A similar configuration needs to be applied for `foo.ns1` in `cluster2`:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: remote-endpoint-for-foo-in-cluster2
+spec:
+  hosts:
+  - foo.ns1.svc.cluster.local
+  ports:
+  # Add all ports from foo service
+  - number: 80
+    name: http
+    protocol: HTTP
+  resolution: STATIC
+  location: MESH_INTERNAL
+  endpoints:
+  - address: <IPofCluster1IngressGateway>
+    ports:
+      http: 15443 #Map everything to 15443 of ingress gateway
+      tcp: 15443
+{{< /text >}}
+
+The gateway for port 15443 has been configured and installed using the
+Istio installation step you saw earlier. Traffic entering port 15443 will
+be load balanced among pods of the appropriate internal service (in this
+case, `bar.ns2`).
+
+> Do not add any Gateway configuration for port 15443.
+
+## Automation
+
+The entire setup above can be automated by replicating Istio configuration,
+service objects, namespaces across all clusters in the mesh. In addition,
+the automation system should generate service entries for each replicated
+service, with endpoints containing the IP address of the gateway in every
+cluster that has pods of the service in question.
+
+## Advanced - Distribute pods of a service across multiple clusters
```

Comment:
I think this doc should stick to the setup basics. There should be one or more example documents that go though some of the more advanced scenarios

Reply:
I generally agree but this example doesn't really show anything about Istio's differentiators from a regular ingress.

Reply:
agreed - we have a slew of other parts of the documentation that can show off the great work in the api.

Reply:
I agree it would be better to remove this section, but ...

I think it would still be good to mention the point that pods could also be distributed across the 2 clusters but that the traffic distribution would be skewed. Maybe put it in the previous section where the ServiceEntry is created, right after the part that says:

`The configuration above will result in all traffic in cluster1 for bar.ns2.svc.cluster.local for any port to be routed to the endpoint <IPofCluster2IngressGateway>:15443 over a mTLS connection.`

Reply:
Ha.. @sdake  @rcaballeromx -- I think you're kind of making the same point as I am.  Kind of my point about this content is actually that all this info can be figured out from the gateway docs (TLS info) and general exposing external services docs (k8s and istio docs).  This is just an example.  The fact that there's 2 clusters involved is coincidental.  If we're going to give a multicluster example let's make it representative of what folks would want to actually do with Istio v. what they can already do with a normal k8s ingress.

Reply:
Lol.. You definitely cannot do this with ingress. There is a special envoy filter configuration for this mcgateway deployed as part of the helm chart, just for this purpose. I didn't want to talk about the complexity of that stuff in this doc and scare people away. I can add a pointer to it though, for interested folks.

Reply:
@rshriram   Oh ok... I didn't catch that there was a special SNI filter on the mcgateway in helm.  So, the SNI routing and Istio features on identity are the important value add here, right?  Rereading the setup instructions requirement of "Enable mTLS globally" for usage of mc gateways kind of hints at that but maybe a blurb calling out the SNI gateway functionality more might add value in this doc.

Reply:
@tiswanso It's true that folks can "figure out" how to do this from other docs. The point of this content is to make it easier by not having people "figure it out". My edits are made in that spirit: easy, clear, and useful. IMO this content can be all three.

Reply:
@rcaballeromx  Sorry, I was thinking this is very similar to what's already documented for a mTLS ingress-gateway (https://istio.io/docs/tasks/traffic-management/secure-ingress/#configure-a-mutual-tls-ingress-gateway) and just the client side is another cluster's service that gets the istio gateway as the endpoint (which is standard k8s external service stuff).  Now I see the difference where the Istio ServiceEntry is what sets up the endpoint info and the mcgateway envoy SNI filter is deployed.